syntax = "proto3";

option java_package = "ca.eandb.jmist.proto";
option java_outer_classname = "CameraProtos";

message PinholeCamera {
  double angle_x = 1;
  double angle_y = 2;
}

message OrthographicCamera {
  double scale_x = 1;
  double scale_y = 2;
}

message Camera {
  repeated double world_to_view = 1;

  oneof parameters {
    PinholeCamera pinhole_camera = 2;
    OrthographicCamera orthographic_camera = 3;
  }
}
